client/core/trade.go

```diff
@@ -1153,12 +1155,17 @@ func (t *trackedTrade) swapMatches(matches []*matchTracker) error {
 	} else {
 		t.coins[hex.EncodeToString(change.ID())] = change
 		t.metaData.ChangeCoin = []byte(change.ID())
+		log.Debugf("Saving change coin %v (%v) to DB for order %v",
+			coinIDString(fromAsset.ID, t.metaData.ChangeCoin), fromAsset.Symbol, t.ID())
 	}
 	t.change = change
-	log.Debugf("Saving change coin %v (%v) to DB for order %v",
-		coinIDString(fromAsset.ID, t.metaData.ChangeCoin), fromAsset.Symbol, t.ID())
 	t.db.UpdateOrderMetaData(t.ID(), t.metaData)
 
+	// Workaround for server recording match ack sig, to avoid an 'init' retry.
+	if includesMakerSwap {
+		time.Sleep(250 * time.Millisecond)
+	}
```

Comment:
Right, and this was one of the first things I had in mind for the dialog.  I can update the comment since I shortened it when moving the sleep to this location